classes/woocommerce-seo.php

```diff
@@ -536,17 +536,35 @@ public function footer_js() {
 	 * @return void
 	 */
 	public function set_yoast_columns_hidden_by_default( $current_screen ) {
-		$user_id = get_current_user_id();
-
-		if (
-			$current_screen->id !== 'edit-product' ||
-			get_user_option( 'wpseo_woo_columns_hidden_default', $user_id ) === '1'
-		) {
+		// Don't do anything if we're not not on the edit products page.
+		if ( $current_screen->id !== 'edit-product' ) {
 			return;
 		}
 
+		$yoast_hidden_columns_old_defaults = [
+			'wpseo-title',
+			'wpseo-metadesc',
+			'wpseo-focuskw',
+		];
+
 		$user_hidden_columns = get_hidden_columns( $current_screen );
 
+		$user_hidden_yoast_columns = array_filter( $user_hidden_columns, [ $this, 'filter_yoast_columns' ] );
+
+		$is_old_default = ( count( array_diff( $yoast_hidden_columns_old_defaults, $user_hidden_yoast_columns ) ) === 0 ) ? true : false;
```

Comment:
No need for the tertiary, `===` already returns a boolean.
```suggestion
		$is_old_default = ( count( array_diff( $yoast_hidden_columns_old_defaults, $user_hidden_yoast_columns ) ) === 0 );
```